storyblok/app-extension-auth
# feat!: new implementation of sessionCookieStore

## What

A new implementation of `sessionCookieStore`. It is based on the old one, with the following changes:

- The `autoRefresh` option in `sessionCookieStore.get` is removed since we've never used it. This is a breaking change, We can reintrodice it later if we need it.
- There are four crud operations for getting, setting, and removing app session cookies: `get`, `getAll`, `put`, and `remove`. The implementation of these are now moved to dedicated files; `getSession`, `getAllSessions`, `putSession`, `removeSession`.
- Created a `utils` module that is only used by the `crud` module.
- The utility function `matches` was renamed to `keysEquals` and moved to its own file.
- The utility function `toKeys` was renamed to `keysFromQuery` and moved to its own file.
- `getSignedCookie` and `setSignedCookie` were changed so that they no longer depend on Node.js (`http` module)

## Why

In the `session` module, the only point where there is a dependency on Node.js is in `sessionCookieStore.ts`. All the other functions accepts functions for reading and writing string values (`GetCookie` and `SetCookie`), but they're actually completely agnostic to how these values are stored. This has two benefits that we will be able to leverage in the future:

- It will become easy to make the library framework agnostic.
- It will become easy to offer alterative ways of storage. It will be better if the `accessToken` is securely stored in a database instead of in a cookie (like we do with the deployments apps).

The next step will be to replace grant with our own, custom request handlers.

## How to test

Run the [Nextjs starter project](https://github.com/storyblok/custom-app-examples/tree/main/app-nextjs-starter) locally.

```
yarn dev
```

 Check out this branch and build the project.

```
yarn build
```

Link the packages with:

In the library:

```shell
yarn link
```

and in the template

```shell
yarn link @storyblok/app-extension-auth
```

In the browser, clear the cookies and reload the app. Everything should work just as before without any modification to the code.
